plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -54,58 +54,98 @@ function perflab_query_plugin_info( string $plugin_slug ) {
 		)
 	);
 
+	$has_errors = false;
+	$plugins    = array();
+
 	if ( is_wp_error( $response ) ) {
-		return new WP_Error(
-			'api_error',
-			sprintf(
-				/* translators: %s: API error message */
-				__( 'Failed to retrieve plugins data from WordPress.org API: %s', 'performance-lab' ),
-				$response->get_error_message()
-			)
+		$plugins[ $plugin_slug ] = array(
+			'error' => array(
+				'code'    => 'api_error',
+				'message' => sprintf(
+					/* translators: %s: API error message */
+					__( 'Failed to retrieve plugins data from WordPress.org API: %s', 'performance-lab' ),
+					$response->get_error_message()
+				),
+			),
 		);
-	}
 
-	// Check if the response contains plugins.
-	if ( ! ( is_object( $response ) && property_exists( $response, 'plugins' ) ) ) {
-		return new WP_Error( 'no_plugins', __( 'No plugins found in the API response.', 'performance-lab' ) );
-	}
+		foreach ( perflab_get_standalone_plugins() as $standalone_plugin ) {
+			$plugins[ $standalone_plugin ] = $plugins[ $plugin_slug ];
+		}
 
-	$plugins      = array();
-	$plugin_queue = perflab_get_standalone_plugins();
+		$has_errors = true;
+	} elseif ( ! ( is_object( $response ) && property_exists( $response, 'plugins' ) ) ) {
```

Comment:
Cognitively easier to remove the negation around the whole clause.
```suggestion
	} elseif ( ! is_object( $response ) || ! property_exists( $response, 'plugins' ) ) {
```